plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java

```diff
@@ -123,25 +141,89 @@ public void onMapReady(final MapboxMap mapboxMap) {
+          // Initialize with the marker's current coordinates.
           makeReverseGeocodingSearch();
         }
+        bindListeners();
 
-        PlacePickerActivity.this.mapboxMap.addOnCameraMoveStartedListener(PlacePickerActivity.this);
-        PlacePickerActivity.this.mapboxMap.addOnCameraIdleListener(PlacePickerActivity.this);
+        if (options != null && options.includeDeviceLocationButton()) {
+          enableLocationComponent(style);
+        } else {
+          userLocationButton.hide();
+        }
       }
     });
   }
 
+  private void adjustCameraBasedOnOptions() {
+    if (options != null) {
+      if (options.startingBounds() != null) {
+        mapboxMap.moveCamera(CameraUpdateFactory.newLatLngBounds(options.startingBounds(), 0));
+      } else if (options.statingCameraPosition() != null) {
+        mapboxMap.moveCamera(CameraUpdateFactory.newCameraPosition(options.statingCameraPosition()));
+      }
+    }
+  }
+
+  @SuppressWarnings( {"MissingPermission"})
+  private void enableLocationComponent(@NonNull Style loadedMapStyle) {
+    // Check if permissions are enabled and if not request
+    if (PermissionsManager.areLocationPermissionsGranted(this)) {
+
+      // Get an instance of the component
+      LocationComponent locationComponent = mapboxMap.getLocationComponent();
+
+      // Activate with options
+      locationComponent.activateLocationComponent(
+        LocationComponentActivationOptions.builder(this, loadedMapStyle).build());
+
+      // Enable to make component visible
+      locationComponent.setLocationComponentEnabled(true);
+
+      // Set the component's camera mode
+      locationComponent.setCameraMode(CameraMode.NONE);
+
+      // Set the component's render mode
+      locationComponent.setRenderMode(RenderMode.NORMAL);
+
+      addUserLocationButton();
+    } else {
+      permissionsManager = new PermissionsManager(this);
+      permissionsManager.requestLocationPermissions(this);
+    }
+  }
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+    permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults);
+  }
+
+  @Override
+  public void onExplanationNeeded(List<String> permissionsToExplain) {
+    Toast.makeText(this, R.string.mapbox_plugins_place_picker_user_location_permission_explanation,
+      Toast.LENGTH_LONG).show();
+  }
+
+  @Override
+  public void onPermissionResult(boolean granted) {
+    if (granted) {
+      mapboxMap.getStyle(new Style.OnStyleLoaded() {
+        @Override
+        public void onStyleLoaded(@NonNull Style style) {
+          if (options != null && options.includeDeviceLocationButton()) {
+            enableLocationComponent(style);
+          }
+        }
+      });
+    } else {
+      Toast.makeText(this, R.string.mapbox_plugins_place_picker_user_location_permission_not_granted,
+        Toast.LENGTH_LONG).show();
+      finish();
```

Comment:
I don't think we should finish the picker activity when permissions are not granted, we should rather just hide the button.

Reply:
Fixed